dvc/ignore.py

```diff
@@ -90,3 +91,39 @@ def __call__(self, root, dirs, files):
             dirs, files = ignore(root, dirs, files)
 
         return dirs, files
+
+
+class CleanTree(BaseTree):
+    def __init__(self, tree):
+        self.tree = tree
+
+    @cached_property
+    def dvcignore(self):
+        return DvcIgnoreFilter(self.tree)
+
+    @property
+    def tree_root(self):
+        return self.tree.tree_root
+
+    def open(self, path, mode="r", encoding="utf-8"):
+        return self.tree.open(path, mode, encoding)
+
+    def exists(self, path):
+        return self.tree.exists(path)
+
+    def isdir(self, path):
+        return self.tree.isdir(path)
+
+    def isfile(self, path):
+        return self.tree.isfile(path)
```

Comment:
Need to check that path is not dvcignored to be entirely safe.

Reply:
I don't think that we should check if its ignored, a path can be both file and be ignored at the same time. What is the reasoning for that?

Reply:
Because if file is ignored then it should not be visible, i.e. tree should pretend it doesn't exist. This is not an issue for now because we only open files that we walk, probably. 

I see at least one error `dvc pipeline show ignored.dvc` will actually instantiate this ignored stage, and we will get a puzzling `KeyError` later.

We may postpone this though because this might be solved in various ways.